src/SelfResponse.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace UKFast\SDK;
+
+class SelfResponse
+{
+    /**
+     * @var string
+     */
+    protected $id;
+
+    /**
+     * @var string
+     */
+    protected $location;
+
+    /**
+     * SelfResponse constructor.
+     * @param $response
+     */
+    public function __construct($response)
+    {
+        $this->id = $response->data->id;
+        $this->location = $response->meta->location;
+    }
+
+    /**
+     * @return int
+     */
+    public function getId()
+    {
+        return $this->id;
+    }
+
+    /**
+     * @return string
+     */
+    public function getLocation()
```

Comment:
I would maybe do it similar to the Page stuff where it can make the request for you and serialize it. E.g.

```php
$self = $pss->requests()->create($request);
$request = $self->get(); // HTTP request happens here
```

Reply:
What do you think @gavtaylor ?